library/std/src/sys/unix/process/process_unix.rs

```diff
@@ -479,7 +479,23 @@ impl ExitStatus {
     }
 
     pub fn signal(&self) -> Option<i32> {
-        if !self.exited() { Some(libc::WTERMSIG(self.0)) } else { None }
+        if libc::WIFSIGNALED(self.0) { Some(libc::WTERMSIG(self.0)) } else { None }
```

Comment:
What's the difference between `exited()` and `IFSIGNALED`?

Reply:
`!exited()` is `!WIFEXITED`, which is not the same as `WIFSIGNALED`.  That is, `WIFEXITED` and `WIFSIGNALED` are not exhaustive.  Some wait statuses are `!WIFSIGNALED && !WIFEXITED`.

The commit message for that commit explains this in a bit more detail, with the example of `WIFSTOPPED`